File: leetCode/228.summary-ranges.cpp
/*
 * @lc app=leetcode id=228 lang=cpp
 *
 * [228] Summary Ranges
 */
#include <bits/stdc++.h>
using namespace std;
// @lc code=start
class Solution
{
public:
  vector<string> summaryRanges(vector<int> &nums)
  {
    vector<string> ans;
    int n = nums.size();
    if (n == 0)
      return ans;
    int a = nums[0];
    for (int i = 0; i < n; i++)
    {
      if (nums[i + 1] != nums[i] + 1 || i == n - 1)
      {
        if (nums[i] != a)
          ans.push_back(to_string(a) + "->" + to_string(nums[i]));
        else
          ans.push_back(to_string(a));
        if (i != n - 1)
          a = nums[i + 1];
      }
    }
    return ans;
  }
};
// @lc code=end

File: leetCode/228.summary-ranges.py
#
# @lc app=leetcode id=228 lang=python3
#
# [228] Summary Ranges
#

from ast import List
# @lc code=start
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if not nums:
            return[]
        res = []
        a = nums[0]
        for i in range(0, len(nums)):
            if i == len(nums) - 1 or nums[i] + 1 != nums[i + 1]:
                if a == nums[i]:
                    res.append(str(a))
                else:
                    res.append(str(a) + '->' + str(nums[i]))
                if i < len(nums) - 1:
                    a = nums[i + 1]
        return res
# @lc code=end
